experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,296 @@
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ...
+```
+
+##### length u32, little endian
+
+length of the entire message, including this field
+
+##### invocation_id, u32, little endian
+
+MUST match the invocation_id of the relevant Request Message
+
+##### status_code, u32, little endian
+
+response status as a
+[gRPC status code](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)
+
+##### padding, 4 bytes
+
+reserved to maintain 8 byte alignment. MUST be unset and ignored
+
+##### body, variable length byte array
+
+For responses with an "Ok" status code the body MUST be the byte encoded method
+return value. Else the body MUST be a utf-8 encoded developer-facing error
+message.
+
+#### Sending Messages
+
+To send a message, it MUST first be encoded into frames. The resulting frame set
+MUST be sent as an uninterrupted sequence.
+
+#### Receiving Messages
+
+To receive a message the recipent incrementally reads frames from the
+communication channel. The recipent MAY start parsing a message before all of
+its frames have been received.
+
+#### Encoding Messages into Frames
+
+The byte representation of a message MUST be chunked into a set of frame bodies.
+The last frame body of the resulting set MAY be shorter than 63,936 bytes (the
+maximum size of a frame body). Any preceding frame bodies MUST be exactly 63,936
+bytes in length.
+
+A frame MUST then be formed from each frame body.
+
+Messages with a length of 63,936 bytes or less MUST be encoded as a single
+frame. In this case the body of the resulting frame contains the entire message.
+
+##### Examples for Illustration
+
+#### Decoding Frames into Messages
+
+Messages are parsed from Frames by appending the frame bodies in the order they
+are received.
+
+### Frames
+
+Messages are sent and received as frames. Each message is fragmented into a set
+of one or more frames.
+
+#### Frame Fields
+
+A byte encoded frame MUST consist of the following fields:
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                            padding                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|             length            |             flags             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                         body bytes...                         +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ...
+```
+
+##### padding, 4 bytes
+
+for future use
+
+##### length, u16, little endian
+
+Length of the entire frame, including this field. The length of a frame MUST NOT
+exceed 64000 bytes
+
+##### flags, two bytes, little endian
```

Comment:
If we treat them as separate bytes (which I think is a good idea), then endianness does not really apply, does it? They are just two consecutive bytes (and it's clear that [0] comes before [1])

Reply:
I think we can remove it, since we indicate LSB and MSB in the flag section later. If represented as seperate bytes the order should be clear :)